src/analysis/vector/qgsgeometrysnappersinglesource.cpp

```diff
@@ -0,0 +1,345 @@
+/***************************************************************************
+  qgsgeometrysnappersinglesource.cpp
+  ---------------------
+  Date                 : May 2018
+  Copyright            : (C) 2018 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsgeometrysnappersinglesource.h"
+
+#include "qgsfeatureiterator.h"
+#include "qgsfeaturesink.h"
+#include "qgsfeaturesource.h"
+#include "qgsfeedback.h"
+#include "qgsgeometrycollection.h"
+#include "qgsgeometryutils.h"
+#include "qgslinestring.h"
+#include "qgspolygon.h"
+#include "qgsspatialindex.h"
+
+//! record about vertex coordinates and index of anchor to which it is snapped
+typedef struct
```

Comment:
This is a rather old-school approach to defining structs! ;)

Reply:
One can nearly smell the old wood :-) but if you prefer more modern style, I can change that